File: UL/System/Int32.cs
namespace System
{
    public struct Int32
    {
        public const Int32 MaxValue = 0x7FFFFFFF;
        public const Int32 MinValue = 0x80000000;


        public extern static Int32 Parse(string value);

        public extern override string ToString();

        public static bool TryParse(string value, out Int32 v)
        {
            try
            {
                v = Parse(value);
                return true;
            }
            catch (Exception e)
            {
                v = 0;
                return false;
            }
        }

        public extern static Int32 operator +(Int32 a, Int32 b);
        public extern static Int32 operator -(Int32 a, Int32 b);
        public extern static Int32 operator *(Int32 a, Int32 b);
        public extern static Int32 operator /(Int32 a, Int32 b);
        public extern static Int32 operator %(Int32 a, Int32 b);
        public extern static Int32 operator &(Int32 a, Int32 b);
        public extern static Int32 operator |(Int32 a, Int32 b);
        public extern static bool operator >(Int32 a, Int32 b);
        public extern static bool operator <(Int32 a, Int32 b);
        public extern static bool operator ==(Int32 a, Int32 b);
        public extern static bool operator !=(Int32 a, Int32 b);

        public extern static Int32 operator <<(Int32 a, int b);
        public extern static Int32 operator >>(Int32 a, int b);

        public extern static Int32 operator ++(Int32 a);
        public extern static Int32 operator --(Int32 a);
        public extern static Int32 operator ~(Int32 a);

        public extern static Int32 operator +(Int32 a);
        public extern static Int32 operator -(Int32 a);

        //public extern static Int32 operator !(Int32 a);

        public static implicit operator Int64(Int32 v);
        public static explicit operator Int16(Int32 v);
        public static implicit operator Single(Int32 v);
        public static implicit operator Double(Int32 v);
        public static explicit operator Byte(Int32 v);
        public static explicit operator SByte(Int32 v);
        public static explicit operator UInt16(Int32 v);
        public static explicit operator UInt32(Int32 v);
        public static explicit operator UInt64(Int32 v);
    }
}

File: UL/CppApp/System/Int32.cpp
#include "stdafx.h"
#include "System\Int32.h"
#include "System\ValueType.h"
#include "System\String.h"
#include "System\Boolean.h"
#include "System\Int64.h"
#include "System\Int16.h"
#include "System\Single.h"
#include "System\Double.h"
#include "System\Byte.h"
#include "System\SByte.h"
#include "System\UInt16.h"
#include "System\UInt32.h"
#include "System\UInt64.h"
#include "System\Exception.h"
#include "Int32_ExtCpp.h"
System::Int32 System::Int32::MaxValue=0x7FFFFFFF;
System::Int32 System::Int32::MinValue=0x80000000;
System::Boolean System::Int32::TryParse(Ref<System::String>  value,System::Int32 & v)
{
	try
	{
		v = Parse(value);
		return true;
	}
	catch(System::Exception e)
	{
		v = 0;
		return false;
	}
}
